libs/vkd3d/vkd3d_private.h

```diff
@@ -239,6 +239,75 @@ HRESULT vkd3d_fence_worker_start(struct vkd3d_fence_worker *worker,
 HRESULT vkd3d_fence_worker_stop(struct vkd3d_fence_worker *worker,
         struct d3d12_device *device);
 
+#define VKD3D_VA_BLOCK_SIZE_BITS (20)
+#define VKD3D_VA_BLOCK_SIZE (1ull << VKD3D_VA_BLOCK_SIZE_BITS)
+#define VKD3D_VA_LO_MASK (VKD3D_VA_BLOCK_SIZE - 1)
+
+#define VKD3D_VA_BLOCK_BITS (20)
+#define VKD3D_VA_BLOCK_COUNT (1ull << VKD3D_VA_BLOCK_BITS)
+#define VKD3D_VA_BLOCK_MASK (VKD3D_VA_BLOCK_COUNT - 1)
+
+#define VKD3D_VA_NEXT_BITS (12)
+#define VKD3D_VA_NEXT_COUNT (1ull << VKD3D_VA_NEXT_BITS)
+#define VKD3D_VA_NEXT_MASK (VKD3D_VA_NEXT_COUNT - 1)
+
+struct vkd3d_unique_resource;
+
+struct vkd3d_va_entry
+{
+    VkDeviceAddress va;
```

Comment:
Since we're doing atomic load/store to this 64-bit, we need to ensure it's 8 byte aligned on 32-bit targets.

Reply:
How do we do that? There's no alignas in C.

Reply:
Turns out we have DECLSPEC_ALIGN. Fixed.